drivers/lm75/lm75.c

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_lm75
+ *
+ * @{
+ * @file
+ * @brief       Driver for the LM75 temperature sensor.
+ *
+ * A general driver for the lm75 temperature sensor including support for the
+ * lm75a and tmp1075 sensors as well.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "lm75.h"
+#include "lm75_regs.h"
+#include "lm75_params.h"
+#include "log.h"
+#include "kernel_defines.h"
+
+#include <stdint.h>
+#include <byteorder.h>
+#include "xtimer.h"
+
+#define I2C_BUS             dev->lm75_params.i2c_bus
+#define I2C_ADDR            dev->lm75_params.i2c_addr
+
+
+int lm75_init(lm75_t *dev, const lm75_params_t *params) {
+
+    dev->lm75_params = *params;
+    uint8_t data;
+    uint8_t config = (params->shutdown_mode) | (params->tm_mode << 1) | (params->polarity << 2) | (params->fault_q << 3);
+
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    /* read the device ID register of the TMP1075 sensor to see if the sensor truly is a TMP1075 */
+    #if MODULE_TMP1075
```

Comment:
```suggestion
    if (IS_USED(MODULE_TMP1075)) {
```

Reply:
I don't understand the reasoning behind this change. Could you explain it to me, please?

Reply:
This way everything gets compiled but the optimizer will throw out the branches that are always false.
The idea is to benefit from the static analysis of the compiler. If the predecessor removes removes the code already we can't spot problems in those dead branches early.